### ISSUE TITLE
keyboard not focus on the correct window  

### ISSUE THREAD

**Issue by dingiso** [hyprwm/Hyprland#5926]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit fe7b748eb668136dd0558b7c8279bfcd7ab4d759  (props: bump version to 0.39.1).
Date: Tue Apr 16 16:01:03 2024
Tag: v0.39.1, commits: 4460

flags: (if any)


System Information:
System name: Linux
Node name: ArchLu
Release: 6.8.9-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Thu, 02 May 2024 17:49:46 +0000


GPU information: 
00:02.0 VGA compatible controller [0300]: Intel Corporation Raptor Lake-S GT1 [UHD Graphics 770] [8086:a780] (rev 04) (prog-if 00 [VGA controller])
01:00.0 VGA compatible controller [0300]: NVIDIA Corporation GP104 [GeForce GTX 1080] [10de:1b80] (rev a1) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:
```

</details>

### Bug or Regression?

Bug

### Description

When I switch workspace using keybindings, the cursor and the focus appears right ( since the border of the window is highlight). However, when I type some words, the screen has no response. Especially when I use tabbed layout, the words I typed always show up in the last tab (the right most of the tab bar) instead the tab been displayed. 

I have try it on non tabbed layout, it sometimes reproduce the problem, but not every time. 

Currently, I can overcome it temporarily by switch in the tabs for several times, or new and close a terminal if switch does not fix it. 

### How to reproduce

Switch workspace using keybindings, and type something.

**Comment by MightyPlaza** [comment]
does the window that keeps the focus have an open pop up, if so it's a dupe of https://github.com/hyprwm/Hyprland/issues/4848 and others

**Comment by vaxerski** [comment]
it is a dupe probably of the 12 other dupes cuz 0.39.1. It's fixed in 0.40

**Comment by ktech0** [comment]
I'm encountering the same issue where I can't type any text using the input method, nor can I switch between input methods. This problem arises when I switch workspaces very quickly, whether using gestures or shortcuts.
